tests/queries/0_stateless/02944_variant_as_if_multi_if_result.sql

```diff
@@ -0,0 +1,65 @@
+set allow_experimental_analyzer=0; -- The result type for if function with constant is different with analyzer.
```

Comment:
but we need to make it work with analyzer, don't think it's a good idea to introduce new code that doesn't work with it

Reply:
Thanks for giving context, as long as you confirmed it with analyzer boss there shouldn't be an issue 😄